### PR TITLE
[5.2] Capability to specify index algorithm

### PR DESCRIPTION
**Pull request by mschelstastic**
This allows specifying the index algorithm for MySQL and PostgreSQL

### CODE REVIEW

File: src/Illuminate/Database/Schema/Blueprint.php [hunk 1]
```diff
@@ -342,11 +342,12 @@ public function rename($to)
      *
      * @param  string|array  $columns
      * @param  string  $name
+     * @param  string  $algorithm
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Schema/Blueprint.php]: string|null

File: src/Illuminate/Database/Schema/Blueprint.php [hunk 2]
```diff
@@ -354,11 +355,12 @@ public function primary($columns, $name = null)
...
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Schema/Blueprint.php]: same

File: src/Illuminate/Database/Schema/Blueprint.php [hunk 3]
```diff
@@ -366,11 +368,12 @@ public function unique($columns, $name = null)
...
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Schema/Blueprint.php]: etc

**Comment by taylorotwell** [comment]
Thanks! I know someone in Slack said to send this to 5.2 but since it does change a method signature it technically has to go to `master` branch. If you send it over there everything else looks pretty good.

**Comment by chelseau** [comment]
Thanks! Updated the phpdocs and sent #12776

**Pull request closed without merging**